src/core.jl

```diff
@@ -314,8 +311,30 @@ julia> common_neighbors(g, 1, 4)
  3
 ```
 """
-common_neighbors(g::AbstractGraph, u::Integer, v::Integer) =
-    intersect(neighbors(g, u), neighbors(g, v))
+function common_neighbors(g::AbstractGraph, u::Integer, v::Integer)
+    common = Vector{eltype(g)}()
+    u_nbrs, v_nbrs = neighbors(g, u), neighbors(g, v)
+    i, j = 1, 1
+    u_len, v_len = length(u_nbrs), length(v_nbrs)
+
+    if (u_nbrs[end] < v_nbrs[1]) || (u_nbrs[1] > v_nbrs[end])
```

Comment:
I think this will be out of bounds, when a vertex has 0 neighbors.

Reply:
Fixed it, I don't know why I miss the corner cases :man_facepalming:

Reply:
I could place to think about what corner cases are, is when you are writing tests.